### DISCUSSION TITLE
Complete noob: RPI4/RPI5-to-RP2040 Scorpio Wiring

### DISCUSSION THREAD

**Discussion by markgabrang** [awawa-dev/HyperSerialPico#30] [Q&A] [answered]
Hello @awawa-dev, thank you for what you've done for the community!
I'm a complete newbie to electronics, and I ordered a RP2040 SCORPIO; I'm having trouble with the wiring.

If my understanding is correct, does it mean that there's no need to connect the RPI4 GPIO to the RP2040 GPIO:
RPI4 > USB > RP2040 (GND + DATA)
RP2024 (GND) > HD107S (GND)
RP2024 (GPIO3) > HD107S (DI)
RP2024 (GPIO2) > HD107S (CI)

Does the RP2040 really only need the USB connection from the RPI4 or RPI5?

Thank you so much!

**Comment by awawa-dev** [comment, marked as answer]
Hi
> Does the RP2040 really only need the USB connection from the RPI4 or RPI5?

Yes, only USB is needed from Rpi to RP2040 to power it up and for sending data using HyperSerialPico.

> If my understanding is correct, does it mean that there's no need to connect the RPI4 GPIO to the RP2040 GPIO:

There is another project that is using GPIO instead of USB: HyperSPI. But HyperSerialPico is using only USB.

>RP2024 (GPIO3) > HD107S (DI)
RP2024 (GPIO2) > HD107S (CI)

It's for generic pico and not for Adafruit RP2040 SCORPIO. That board is using custom pinout because of the level shifter. It's explained here: https://github.com/awawa-dev/HyperSerialPico?tab=readme-ov-file#how-to-flash-it

```Due to the often neglected issue of the required 5 volt logic necessary for the correct operation of the supported LEDs, we want to promote rp2040 boards with a built-in 3.3V to 5V level shifter and we provide firmware compiled to their GPIO specifications. You can find these firmware & >>>short pin-out manuals <<< in the archives: Adafruit_Feather_RP2040_Scorpio.zip, Adafruit_ItsyBitsy_2040.zip, Pimoroni_Plasma_2040.zip, Pimoroni_Plasma_Stick_2040_W.zip. ```

**Reply by markgabrang** [reply]
Hi! Thank you for the very quick reply, I was half-expecting to get an answer in a few days.

I did buy the Adafruit version, so the connections become:
Adafruit SCORIO RP2024 (GPIO16) > HD107S (DI)
Adafruit SCORIO RP2024 (GPIO23) > HD107S (CI)

Again, thank you so much as I'm having a lot of trouble wrapping my head around this project.

**Reply by awawa-dev** [reply]
No that's wrong :wink: In Adafruit_Feather_RP2040_Scorpio.zip you can find Firmwares_for_Adafruit_Feather_RP2040_Scorpio.txt file and inside of it:
```
Neopixel is using GPIO16(OUTPUT_DATA_PIN) on output 0.
SPI is using spi0 interface pins: GPIO19(OUTPUT_SPI_DATA_PIN) and GPIO18(OUTPUT_SPI_CLOCK_PIN) on output 3 and 2 respectively.
```

**Reply by markgabrang** [reply]
Oof! My bad!

So.. it's the physical pins numbered 3 and 2 on this image?:
https://cdn-learn.adafruit.com/assets/assets/000/117/535/large1024/adafruit_products_feather-rp2040-scorpio-prettypins.png?1673545808

**Reply by awawa-dev** [reply]
NEOPIXEL3 and NEOPIXEL2 at the bottom

**Reply by markgabrang** [reply]
That makes everything perfectly clear, thank you so much! 🫡
